### ISSUE TITLE
Optimize recall and accuracy for a sub-class in multi-class classification

### ISSUE THREAD

**Issue by chalbiophysics** [tensorflow/tensorflow#39724]
Hi everyone,

Sorry about asking this question but I am a beginner with Tensorflow and Keras.
Here is my question :
- I have annotated 2.000 images of living cells, 8.000 images of dead cells and 8.000 images with nothing. They are labeleled respectively as "Living", "Death" and "Empty". (Images are 299x299x1)
- I have run some models, which basically look like this:

```
model = Sequential([
                    Conv2D(16, (5,5), activation='relu', input_shape=(299,299,1)),
                    MaxPooling2D(pool_size=(2,2)),
                    Conv2D(32, (5,5), activation='relu'),
                    Flatten(),
                    Dense(200),
                    Activation('relu'),
                    Dropout(0.3),
                    Dense(3),
                    Activation('softmax')
])

opt = SGD(lr=0.001)
model.compile(loss = "categorical_crossentropy", 
              optimizer = opt,
              metrics=['accuracy'])
```
- At the end of training, validation accuracy is around 90%. Nevertheless, when I check with confusion matrix, it appears that classification is very good to detect "Empty" images (accuracy almost 99%), but it is less good for separation of "Living" from "Death" images (circa 80%).
- The main purpose for me is to optimize recall and accuracy for classification of "Living" cells. 

=> That's why I am wondering if there is a way to track recall and accuracy for a specific sub-class ('Living" here) like in early-stop.

Thank you for your help !

Kindly !

**Comment by Saduf2019** [comment]
@chalbiophysics 
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged//tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

**Comment by wysohn** [comment]
You can always write your own metrics.

For your code:

```
model.compile(loss = "categorical_crossentropy", 
              optimizer = opt,
              metrics=['accuracy'])
```

Simply define a metric function and calculate the accuracy yourself.

```
def some_metric(y_true, y_pred):
    return ...

model.compile(loss = "categorical_crossentropy", 
              optimizer = opt,
              metrics=[[some_metric]])
```

where y_true being the label, (BATCH, 3) as you have three classes, and y_pred being the predicted value (BATCH, 3)

note, however, that the metric function needs to return the batch of accuracies (BATCH,), not a single value.

**Comment by chalbiophysics** [comment]
Thank you @wysohn !
